sentry-android-core/src/main/java/io/sentry/android/core/gestures/SentryGestureListener.java

```diff
@@ -0,0 +1,238 @@
+package io.sentry.android.core.gestures;
+
+import android.view.GestureDetector;
+import android.view.MotionEvent;
+import android.view.View;
+import android.view.Window;
+import io.sentry.Breadcrumb;
+import io.sentry.IHub;
+import io.sentry.SentryLevel;
+import io.sentry.android.core.SentryAndroidOptions;
+import java.lang.ref.WeakReference;
+import java.util.Collections;
+import java.util.Map;
+import org.jetbrains.annotations.ApiStatus;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+@ApiStatus.Internal
+public final class SentryGestureListener implements GestureDetector.OnGestureListener {
+
+  private final @NotNull WeakReference<Window> windowRef;
+  private final @NotNull IHub hub;
+  private final @NotNull SentryAndroidOptions options;
+
+  private final ScrollState scrollState = new ScrollState();
+
+  public SentryGestureListener(
+    final @NotNull WeakReference<Window> windowRef,
+    final @NotNull IHub hub,
+    final @NotNull SentryAndroidOptions options
+  ) {
+    this.windowRef = windowRef;
+    this.hub = hub;
+    this.options = options;
+  }
+
+  public void onUp(final @NotNull MotionEvent motionEvent) {
+    final View decorView = ensureWindowDecorView("onUp");
+    final View scrollTarget = scrollState.targetRef.get();
+    if (decorView == null || scrollTarget == null) {
+      return;
```

Comment:
`scrollTarget` will be null if the strong reference count went on to be 0 and GC collected it. In that case, this class becomes useless. Is this a valid scenario? Or is this something that can happen during shutdown or some other use case?

Trying to understand why we're going with WeakRef here

Reply:
This is to make sure we are not leaking the underlying activity/fragment. If the ref count becomes 0, this class *should* become useless, as the activity/fragment got destroyed in that case.